watertap/tools/oli_api/credentials.py

```diff
@@ -237,19 +237,22 @@ def set_access_key(self):
             return self.credentials["access_keys"][r]
 
     # TODO: improve header management for class
-    def generate_oliapi_access_key(self):
+    # TODO: possibly save api keys as objects with expiry and other attributes
+    def generate_oliapi_access_key(self, expiry_timecode):
```

Comment:
Either here or further up in the API we should consider automatically converting `datetime.datetime` objects to the appropriate `expiry_timecode`, with a sane default base on the current time (or maybe leaving it off and defaulting this to `None`).